### ISSUE TITLE
Controlled popover targets main window initially instead of a popup window

### ISSUE THREAD

**Issue by GerardasB** [iTwin/iTwinUI#2093]
### Describe the bug (current behavior)

An edge case of #2006
When `Popover` is rendered with `visible={true}` initially, the popover content is still rendered to the main window instead of a popup window to which it is reparented.

### Expected Behavior

Popover content should be rendered to the popup window.

### Link to minimal repro

https://codesandbox.io/p/sandbox/popout-initially-visible-popover-dzd7qy

### Steps To Reproduce

1. Click `Open Window Popup` to open a new popup window
2. Observe that `Popover content` is rendered in the main window **incorrectly**
3. Click on `Target` multiple times to change popover visibility
4. Observe that `Popover content` is now rendered correctly

### Anything else?

I think the `Tooltip` component has analogous behavior.

**Comment by mayank99** [comment]
Hey, your sandbox link is 404.

We can look into this, but in general, I think `Popover`/`Tooltip` should _not_ be rendered with `visible={true}` on first mount. There should almost always be a trigger element that opens the popover/tooltip. What's your use case for this?

**Comment by mayank99** [comment]
> I was trying to re-use the popover component to render&position the element tooltip [iTwin/appui#862](https://github.com/iTwin/appui/issues/862)

Shouldn't that tooltip appear after some user interaction (e.g. click on a point in the canvas) rather than on initial mount?

**Comment by GerardasB** [comment]
> Shouldn't that tooltip appear after some user interaction (e.g. click on a point in the canvas) rather than on initial mount?

Yes, that should usually be the case. However, since the `Tooltip`/`Poporver` is a controlled component, that can be visible initially, it should not be broken on the initial render pass.

I.e. in `ElementTooltip` case the tooltip is rendered and positioned based on which decorator/element is hovered (there's an event emitted with required information from the viewport APIs to show/hide the tooltip). While portalling the tooltip, it is always re-mounted for a new portal container.
To workaround this I'd need to do some hacks and render a separate instance for each window OR make sure to first mount with `visible={false}` and then re-render.

**Comment by mayank99** [comment]
> However, since the `Tooltip`/`Popover` is a controlled component, that can be visible initially, it should not be broken on the initial render pass.

I understand what you're saying, however:
1. This is still a code concern, whereas I'm asking about it from the user's perspective. Your example still sounds like it appears after some user interaction.
2. This is a very difficult problem to solve for all cases. For a normal React component, we can simply let React take care of the rendering. But in the case of portaled elements, we need to wait for the DOM to stabilize, which is sometimes literally impossible (e.g. when doing SSR).

> make sure to first mount with visible={false} and then re-render.

This is not a hack or workaround. This is the canonical way to use these APIs. `Tooltip`/`Popover` should always start out as `visible={false}`, and then after some user interaction should become `visible={true}`. I would even add that conditional rendering might be the wrong way to use these APIs.

**Comment by mayank99** [comment]
This bug is fixed in [3.12.0](https://github.com/iTwin/iTwinUI/releases/tag/%40itwin/itwinui-react%403.12.0), although I would still advise against showing popovers on the initial render. 

Also, it seems like this issue was only applicable to `Popover`, not `Tooltip`. It's worth noting that `Popover` is meant for dialog-like elements that are triggered by click, not hover, so you might be misusing the component here.
